src/test/ui/question-mark-type-infer.stderr

```diff
@@ -1,8 +1,13 @@
-error[E0284]: type annotations needed: cannot resolve `<_ as std::ops::Try>::Ok == _`
-  --> $DIR/question-mark-type-infer.rs:12:5
+error[E0284]: type annotations needed
+  --> $DIR/question-mark-type-infer.rs:12:21
    |
 LL |     l.iter().map(f).collect()?
-   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |                     ^^^^^^^
+   |                     |
+   |                     cannot infer type
+   |                     help: consider specifying the type argument in the method call: `collect::<B>`
```

Comment:
I'm a bit unsure about introducing the letter `B` here. It could be that the name happens to shadow a name that's already in scope for the user, which might be confusing. I wonder if we could write something like `collect::<_>` or is that potentially more confusing? (Especially as it would not be especially helpful if they were to write that exact thing, though the same applies to `B`.)

Reply:
I think the only way to improve on this is not to use `_` but rather to attempt to give a reasonable type that could fit there, but that would be out of scope for this PR. My main objective here is to 1) bring attention to the fact that this method has a type argument and 2) what the turbofish looks like.